app/facades/webhooks.py

```diff
@@ -0,0 +1,71 @@
+import json
+import base64
+from typing import Union
+import os
+from typing import List, Literal
+from aries_cloudcontroller import AcaPyClient
+
+from httpx import AsyncClient, get, HTTPError
+
+
+WEBHOOKS_URL = os.getenv("WEBHOOKS_URL", "http://yoma-webhooks-web:3010")
+ADMIN_API_KEY = os.getenv("ACAPY_ADMIN_API_KEY", "adminApiKey")
+
+topics = Literal[
+    "connections",
+    "issue_credential",
+    "forward",
+    "ping",
+    "basicmessages",
+    "issuer_cred_rev",
+    "issue_credential_v2_0",
+    "issue_credential_v2_0_indy",
+    "issue_credential_v2_0_dif",
+    "present_proof",
+    "revocation_registry",
+]
+
+
+def get_wallet_id_from_client(client: Union[AcaPyClient, AsyncClient]) -> str:
+
+    # eg tenenat_jwt: "eyJ3YWxsZXRfaWQiOiIwMzg4OTc0MC1iNDg4LTRmZjEtYWI4Ni0yOTM0NzQwZjNjNWMifQ"
+    if isinstance(client, AcaPyClient):
+        jwt = client.client.headers["authorization"].split(" ")[1].split(".")[1]
+    elif isinstance(client, AsyncClient):
+        jwt = client.headers.get("x-api-key").split(".")[2]
+    else:
+        jwt = client.tenant_jwt
+
+    # Add padding if required
+    # b64 needs lengths divisible by 4
+    if len(jwt) % 4 != 0:
+        n_missing = 4 - (len(jwt) % 4)
+        jwt = jwt + (n_missing * "=")
+
+    wallet = json.loads(base64.b64decode(jwt))
+    return wallet["wallet_id"]
+
+
+def get_hooks_per_topic_per_wallet(client: AcaPyClient, topic: topics) -> List:
+    wallet_id = get_wallet_id_from_client(client)
+    try:
+        hooks = (get(f"{WEBHOOKS_URL}/{topic}/{wallet_id}")).json()
+        return hooks if hooks else []
+    except HTTPError as e:
+        raise e from e
+
+
+def get_hooks_per_topic_admin(client: AcaPyClient, topic: topics) -> List:
+    """
+    Gets all webhooks for all wallets by topic (default="connections")
+    """
+    try:
+        # Ensure admin key is present
+        assert client.client.headers["x-api-key"] == ADMIN_API_KEY
```

Comment:
With mulitple ACA-Py instances there are multiple admin api keys right?

Reply:
Hm - that is a good question. I didn't think of that. Is that a realistic scenario? and how would the client handle that (when targeting an aca-py instance)? Wouldn't it have to knwo which one to use then somehow (the client I mean)? If this is a problem here, isn't that also a problem in general? Maybe we can have a quick discussion about this. I'm not quite sure

Reply:
This is still outstanding as I'm nto yet sure how to handle this best, but working on it.